pkg/clusterapi/name.go

```diff
@@ -0,0 +1,42 @@
+package clusterapi
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+)
+
+var nameRegex = regexp.MustCompile(`(-)(\d+)$`)
+
+func IncrementName(name string) (string, error) {
+	match := nameRegex.FindStringSubmatch(name)
+	if match == nil {
+		return "", fmt.Errorf(`invalid format of name [name=%s]. Name has to follow regex pattern "(-)(\d+)$", e.g. machinetemplate-cp-1`, name)
+	}
+
+	n, err := strconv.Atoi(match[2])
+	if err != nil {
+		return "", fmt.Errorf("converting object suffix to int: %v", err)
+	}
+
+	return nameRegex.ReplaceAllString(name, fmt.Sprintf("${1}%d", n+1)), nil
```

Comment:
if we change the regex, we can extract the name as the group 1 and just do `ObjectName(match[1], n+1)`